cooldowns/cooldown.py

```diff
@@ -438,3 +439,227 @@ def bucket(self) -> CooldownBucketProtocol:
+                cooldown_id= self.cooldown_id,
+                check= self.check
+            )
+
+        if cooldown_id:
+            utils.shared_cooldown_refs[cooldown_id] = self.cooldown
+
+    async def trigger(self, time_period: Union[float, datetime.timedelta]) -> None:
+        """|coro|
+
+        Trigger the Trigger Cooldown instantly. Has to be awaited.
+
+        Parameters
+        ----------
+        time_period : `Union[float, datetime.timedelta]`
+            The time period that cooldwon will remain triggered.
+        """
+        self.triggered = True
+        self.trigger_cooldown.time_period = (
+            time_period
+            if isinstance(time_period, (float, int))
+            else time_period.total_seconds()
+        )
+
+        # Triggers the Cooldown leaving bucket.current = 0
+        frame = inspect.currentframe().f_back
+        _, _, _, values = inspect.getargvalues(frame)
+        args = tuple(values.values())
+
+        async with self.trigger_cooldown(*args):
+            return None
+
+    
+    def __call__(self, func: Callable) -> Callable:
+        """
+        
+        Called as a decorator.
+
+        Parameters
+        ----------
+        func : `Callable`
+            The function being decorated.
+
+        Returns
+        -------
+        `Callable`
+            Decorator
+
+        Raises
+        ------
+        `RuntimeError`
+            When given function is not coroutine.
+        """
+
+        if not asyncio.iscoroutinefunction(func):
+            raise RuntimeError(
+                f"Expected `func` to be a coroutine, "
+                f"found {func} of type {func.__class__.__name__!r} instead"  # noqa
+            )
+        # Links the cooldowns to the given function.
+        self.cooldown._func = func
+        self.trigger_cooldown._func = func
+
+        @functools.wraps(func)
+        async def inner(*args, **kwargs):
+            use_cooldown = await maybe_coro(self.check, *args, **kwargs)
+            if not use_cooldown:
+                return await maybe_coro(func, *args, **kwargs)
+            
+            self_arg = None
+            if "self" in kwargs:
+                self_arg = kwargs.pop("self")
+
+            # If the cooldown is triggered...
+            if self.triggered:
+                # If still on triggered cooldown...
+                if self.trigger_cooldown.remaining_calls(*args, **kwargs) < 1:
+                    # Runs the Trigger Cooldown.
+                    async with self.trigger_cooldown(*args, **kwargs):
+                        if self_arg:
+                            kwargs["self"] = self_arg
+                            result = await func(*args, **kwargs)
+                        else:
+                            result = await func(*args, **kwargs)
+                    return result
+                # If not, untrigger the cooldown.
+                else:
+                    self.triggered = False
```

Comment:
So if you hit this condition, zero cooldowns run at all..? Seems like unwanted behaviour and you should be falling back to the base cooldown instead

Reply:
No, if hit that condition, the execution continues to the next cooldown, which is the normal cooldown defined. That means, always one cooldown will be activated, or the trigger one, or the normal one.

Here you have a simplified version of the code so you can understand:
```python
if self.triggered:
   # do things. If trigger cooldown is activated, here "return" is done.

# If no trigger cooldown was actually trigger, then no return statement so...
 async with self.cooldown(*args, **kwargs): # <-- Normal cooldown
```

Reply:
Sounds good, indentation just looked funky on my previous browser